graphqlapi/src/main/java/io/stargate/graphql/schema/fetchers/ddl/CreateIndexFetcher.java

```diff
@@ -0,0 +1,104 @@
+/*
+ * Copyright The Stargate Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.stargate.graphql.schema.fetchers.ddl;
+
+import com.google.common.base.Splitter;
+import graphql.schema.DataFetchingEnvironment;
+import io.stargate.auth.AuthenticationService;
+import io.stargate.auth.AuthorizationService;
+import io.stargate.auth.Scope;
+import io.stargate.db.datastore.DataStoreFactory;
+import io.stargate.db.query.Query;
+import io.stargate.db.query.builder.QueryBuilder;
+import io.stargate.db.schema.CollectionIndexingType;
+import io.stargate.db.schema.ImmutableCollectionIndexingType;
+import java.util.Arrays;
+import java.util.List;
+
+public class CreateIndexFetcher extends IndexFetcher {
+  public CreateIndexFetcher(
+      AuthenticationService authenticationService,
+      AuthorizationService authorizationService,
+      DataStoreFactory dataStoreFactory) {
+    super(authenticationService, authorizationService, dataStoreFactory);
+  }
+
+  @Override
+  protected Scope getScope() {
+    return Scope.CREATE;
+  }
+
+  @Override
+  protected Query<?> buildQuery(
+      DataFetchingEnvironment dataFetchingEnvironment,
+      QueryBuilder builder,
+      String keyspaceName,
+      String tableName) {
+    String columnName = dataFetchingEnvironment.getArgument("columnName");
+
+    if (keyspaceName == null) {
+      List<String> parts = Splitter.on('.').splitToList(tableName);
+      if (parts.size() < 2) {
+        throw new IllegalArgumentException("Missing field argument keyspaceName @ 'createIndex'");
+      }
+      keyspaceName = parts.get(0);
+      tableName = parts.get(1);
```

Comment:
Why not, but if we do this here, we should revisit all the table and type fetchers for uniformity. I'm not sure if it's worth it, seems simple enough to pass the keyspace separately.

Reply:
I have removed the optional keyspaceName. 👍